node-metrics/src/service/client_message/mod.rs

```diff
@@ -0,0 +1,245 @@
+use super::client_id::ClientId;
+use serde::{Deserialize, Serialize};
+
+/// [ClientMessage] represents the messages that the client can send to the
+/// server for a request.
+///
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub enum ClientMessage {
+    SubscribeLatestBlock,
+    SubscribeNodeIdentity,
+    SubscribeVoters,
+
+    RequestBlocksSnapshot,
+    RequestNodeIdentitySnapshot,
+    RequestHistogramSnapshot,
+    RequestVotersSnapshot,
+}
```

Comment:
I am not certain I quite understand the question.

These are client side messages that come across a single `WebSocket`, so we are using `WebSocket`s?

If we were to split up each subscribe to its own endpoint, then a client browser would need to maintain 3 open `WebSocket` connections in order to stream the `LatestBlock`, `NodeIdentity`, and `Voters` information.  It seems somewhat wasteful to have 3 persistent open connections per client to the backend, and perhaps this might grow in the future to subscribe to other data, especially when we can facilitate the same functionality through a single `WebSocket`.

An alternative would be for the client to not submit anything in terms of the requests, and then just stream all of the data on connection to the end user.  However, I wanted some flexibility / backwards compatability on this.  If we decide to add a new stream to the API, and it starts getting processed by a down-stream client, there's a possibility it won't recognize the messages, and that could cause down-stream errors.  Additionally, perhaps we want lighter versions of the data being downloaded.  For instance, the `Explorer` main page, Lucca or Charles had asked the question about why they don't update the blocks in real-time.  Well, we could just opt in to the subscription data we want to follow instead of just getting everything upon request.  I feel like opting "in" addresses both of these concerns.

Reply:
That all makes sense, thanks for explaining!